src/services/has-many-getter.js

```diff
@@ -80,26 +80,38 @@ function HasManyGetter(model, association, opts, params) {
         var query = association.find(conditions);
         handlePopulate(query);
 
-        return query;
+        return query.then(function(records) {
+          return [records, recordIds];
+        });
       })
-      .then(function(records) {
-        if (params.sort) {
-          var fieldSort = params.sort;
-          var descending = false;
-
-          if (params.sort[0] === '-') {
-            fieldSort = params.sort.substring(1);
-            descending = true;
-          }
+      .then(function(recordsAndRecordIds) {
+        var records = recordsAndRecordIds[0];
+        var fieldSort = params.sort;
+        var descending = false;
+
+        if (params.sort && (params.sort[0] === '-')) {
+          fieldSort = params.sort.substring(1);
+          descending = true;
+        }
 
+        if (fieldSort && (fieldSort !== 'parentsReferencingOrder')) {
```

Comment:
Is this condition (`fieldSort !== 'parentsReferencingOrder'`) really generic ? Where does the `parentsReferencingOrder` comes from?

Reply:
Yes it is not generic and I removed it. I thought it would help to implemented a descending order in the future, since there is no button for the user to reverse the order.